pkg/api/v20210901preview/openshiftcluster_validatestatic.go

```diff
@@ -201,6 +200,16 @@ func (sv openShiftClusterStaticValidator) validateNetworkProfile(path string, np
 		return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidParameter, path+".podCidr", "The provided pod CIDR '%s' is invalid: must be IPv4.", np.PodCIDR)
 	}
 
+	switch np.SoftwareDefinedNetwork {
```

Comment:
nit: we can make it `if`.

This is not a change request, but I'll write it down what I thought.
I thought we should have the same validation in other versions.
If someone tries to change podCIDR to 100.64.0.0/16 in the old cluster with old SDN, it will get the overlap error, but it should be `PropertyChangeNotAllowed`.
However only this version has the `SoftwareDefinedNetwork` field, so other versions don't have a way to know which SDN is being used.
Changing the fields would be too much effort, so I think we're good to go.